lib/grpc/server.ex

```diff
@@ -107,16 +107,23 @@ defmodule GRPC.Server do
          %{request_mod: req_mod, codec: codec, adapter: adapter, payload: payload} = stream,
          func_name
        ) do
-    {:ok, data} = adapter.read_body(payload)
 
-    case GRPC.Message.from_data(stream, data) do
-      {:ok, message} ->
-        request = codec.decode(message, req_mod)
+    try do
+      {:ok, data} = adapter.read_body(payload)
+
+      case GRPC.Message.from_data(stream, data) do
+        {:ok, message} ->
+          request = codec.decode(message, req_mod)
 
-        call_with_interceptors(res_stream, func_name, stream, request)
+          call_with_interceptors(res_stream, func_name, stream, request)
 
-      resp = {:error, _} ->
-        resp
+        resp = {:error, _} ->
+          resp
+      end
+    rescue
+      err ->
+        report_to_exception_sinks(stream, err, __STACKTRACE__)
```

Comment:
I wonder if we still need this feature given that we now have specific telemetry events that capture the exact data that would be reported to `exception_sinks`. @freshtonic thoughts on this?